### PR TITLE
Enforce boolean `ascending` for `sort_values`

### PR DESCRIPTION
**Pull request by charlesbluca**
It is possible to pass a list of ascending booleans to `sort_values`, which is not yet supported by `_calculate_divisions` (which computes divisions in ascending order), and can cause undefined behavior.

This small PR adds a check that ascending is a boolean before calling `rearrange_by_divisions`; note that this check happens _after_ all single-partition cases are handled, as Pandas can handle lists of ascending booleans.

- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

File: dask/dataframe/shuffle.py [hunk 1]
```diff
@@ -114,6 +114,9 @@ def sort_values(
             sort_function, **sort_function_kwargs
         )
 
+    if not isinstance(ascending, bool):
+        raise ValueError("ascending must be either True or False")
```
Comment (ian-r-rose) [inline dask/dataframe/shuffle.py:118]: Can you indicate in this message that `ascending` as a list of bools is not supported? Just to communicate to the user that it's a dask limitation, not a mistake that they made

**Review by ian-r-rose — COMMENTED** [review]
Thanks @charlesbluca , can you also add a test demonstrating this behavior?

File: dask/dataframe/shuffle.py (hunk 1 above)
Comment (charlesbluca) [reply dask/dataframe/shuffle.py]: Done

**Comment by jsignell** [comment]
This looks nice! Just missing a test like Ian mentioned.

**Comment by jsignell** [comment]
@charlesbluca, I think if you merge or rebase off main, lots of these tests will pass.

**Comment by charlesbluca** [comment]
Thanks @jsignell, merged in the latest

**Comment by jsignell** [comment]
Thanks @charlesbluca!

**Pull request merged**